silo-core/contracts/utils/liquidationHelper/LiquidationHelper.sol

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import {Address} from "openzeppelin5/utils/Address.sol";
+import {IERC20} from "openzeppelin5/token/ERC20/IERC20.sol";
+
+import {IERC3156FlashBorrower} from "../../interfaces/IERC3156FlashBorrower.sol";
+import {IPartialLiquidation} from "../../interfaces/IPartialLiquidation.sol";
+import {ILiquidationHelper} from "../../interfaces/ILiquidationHelper.sol";
+
+import {ISilo} from "../../interfaces/ISilo.sol";
+import {ISiloConfig} from "../../interfaces/ISiloConfig.sol";
+import {IWrappedNativeToken} from "../../interfaces/IWrappedNativeToken.sol";
+
+import "./DexSwap.sol";
+
+interface ISiloConfigHelper {
+    function siloConfig() external view returns (ISiloConfig);
+}
+
+/// @notice LiquidationHelper IS NOT PART OF THE PROTOCOL.
+contract LiquidationHelper is ILiquidationHelper, IERC3156FlashBorrower, DexSwap {
+    using Address for address payable;
+
+    // solhint-disable-next-line var-name-mixedcase
+    bytes32 internal constant _FLASHLOAN_CALLBACK = keccak256("ERC3156FlashBorrower.onFlashLoan");
+
+    /// @dev token receiver will get all rewards from liquidation, does not matter who will execute tx
+    address payable public immutable TOKENS_RECEIVER;
+
+    /// @dev address of wrapped native blockchain token eg. WETH on Ethereum
+    address public immutable NATIVE_TOKEN;
+
+    uint256 private transient _withdrawCollateral;
+    uint256 private transient _repayDebtAssets;
+
+    error NoDebtToCover();
+    error STokenNotSupported();
+
+    /// @param _nativeToken address of wrapped native blockchain token eg. WETH on Ethereum
+    /// @param _exchangeProxy exchange address, where to send swap data on liquidation
+    /// @param _tokensReceiver all leftover tokens (debt and collateral) will be send to this address after liquidation
+    constructor (
+        address _nativeToken,
+        address _exchangeProxy,
+        address payable _tokensReceiver
+    ) DexSwap(_exchangeProxy) {
+        NATIVE_TOKEN = _nativeToken;
+        EXCHANGE_PROXY = _exchangeProxy;
+        TOKENS_RECEIVER = _tokensReceiver;
+    }
+
+    /// @inheritdoc ILiquidationHelper
+    /// @dev entry point for liquidation
+    /// @notice for now we does not support liquidation with sTokens
+    /// on profitable liquidation we will revert because we will not be able to repay flashloan with fee
```

Comment:
Can you clarify this comment? The 'revert' needs to be clarified. Do you mean that it will revert if we cannot repay the flash loan with the fee, or is this about unprofitable liquidation and sTokens?

Reply:
I updated comment.